pdr_backend/analytics/predictoor_dashboard/test/conftest.py

```diff
@@ -42,3 +64,65 @@ def pytest_setup_options():
     options.add_argument("--headless")
     options.add_argument("--disable-gpu")
     return options
+
+
+@pytest.fixture
+def setup_app(
+    tmpdir, _sample_daily_predictions, _sample_payouts_related_with_predictions
+):
+    _clear_test_db(str(tmpdir))
+
+    _prepare_test_db(
+        tmpdir,
+        _sample_payouts_related_with_predictions,
+        table_name=Payout.get_lake_table_name(),
+    )
+
+    ppss, _ = _prepare_test_db(
+        tmpdir,
+        _sample_daily_predictions,
+        table_name=Prediction.get_lake_table_name(),
+    )
+
+    app = Dash(__name__, external_stylesheets=[dbc.themes.BOOTSTRAP])
+    app.config["suppress_callback_exceptions"] = True
+
+    style_css = ""
+    # read the styles.css file and add it to the assets folder
+    # Read the styles.css file
+    with open("pdr_backend/analytics/predictoor_dashboard/assets/styles.css", "r") as f:
+        style_css = f.read()
+
+    # Manually link the CSS file by embedding its contents in a <style> tag
+    app.index_string = f"""
+    <!DOCTYPE html>
+    <html>
+        <head>
+            {{%metas%}}
+            <title>{{%title%}}</title>
+            {{%favicon%}}
+            {{%css%}}
+            <style>{style_css}</style>
+        </head>
+        <body>
+            {{%app_entry%}}
+            <footer>
+                {{%config%}}
+                {{%scripts%}}
+                {{%renderer%}}
+            </footer>
+        </body>
+    </html>
+    """
+
+    app.layout = get_layout()
+    app.lake_dir = ppss.lake_ss.lake_dir
+    app.feeds_data = get_feeds_data_from_db(ppss.lake_ss.lake_dir)
```

Comment:
to avoid possible inconsistencies with this part, let's create a function that takes app and ppss as parameters, called e.g. `setup_predictoor_dashboard` and add predictoors_data, feeds_data, lake_dir, favourite_addresses. Call this function here and where this code is duplicated. I wanted to do this myself at the last round of fixes but I missed it

Reply:
other dashboards have at most one line of setup, so that's why we just copy that setup in the tests, but here I think it's safer to do a function